### PR TITLE
time namespace support

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

File: src/lxc/confile.c [hunk 1]
```diff
@@ -166,110 +168,112 @@ lxc_config_define(proc);
...
+	{ "lxc.mount.entry",                     set_config_mount,                            get_config_mount,                            clr_config_mount,                           },
+	{ "lxc.mount.fstab",                     set_config_mount_fstab,                      get_config_mount_fstab,                      clr_config_mount_fstab,                     },
+	{ "lxc.namespace.clone",                 set_config_namespace_clone,                  get_config_namespace_clone,                  clr_config_namespace_clone,                 },
+	{ "lxc.namespace.keep",                  set_config_namespace_keep,                   get_config_namespace_keep,                   clr_config_namespace_keep,                  },
+	{ "lxc.namespace.time.offset.boot",	 set_config_namespace_time_offset_boot,       get_config_namespace_time_offset_boot,       clr_config_namespace_time_offset_boot,      },
+	{ "lxc.namespace.time.offset.monotonic", set_config_namespace_time_offset_monotonic,  get_config_namespace_time_offset_monotonic,  clr_config_namespace_time_offset_monotonic, },
```
Comment (stgraber) [inline src/lxc/confile.c:227]: Shouldn't that just be `lxc.time.offset.boot` and `lxc.time.offset.monotonic`?

File: src/lxc/confile.c [hunk 2]
```diff
@@ -222,9 +222,9 @@ static struct lxc_config_t config_jump_table[] = {
...
 	{ "lxc.namespace.clone",                 set_config_namespace_clone,                  get_config_namespace_clone,                  clr_config_namespace_clone,                 },
 	{ "lxc.namespace.keep",                  set_config_namespace_keep,                   get_config_namespace_keep,                   clr_config_namespace_keep,                  },
+	{ "lxc.namespace.share",                 set_config_namespace_share,                  get_config_namespace_share,                  clr_config_namespace_share,                 },
 	{ "lxc.namespace.time.offset.boot",	 set_config_namespace_time_offset_boot,       get_config_namespace_time_offset_boot,       clr_config_namespace_time_offset_boot,      },
 	{ "lxc.namespace.time.offset.monotonic", set_config_namespace_time_offset_monotonic,  get_config_namespace_time_offset_monotonic,  clr_config_namespace_time_offset_monotonic, },
-	{ "lxc.namespace.share",                 set_config_namespace_share,                  get_config_namespace_share,                  clr_config_namespace_share,                 },
```
Comment (stgraber) [inline src/lxc/confile.c]: Not sure what that's doing in this commit

File: doc/lxc.container.conf.sgml.in [hunk 3]
```diff
@@ -1806,6 +1806,33 @@ Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
...
+            <option>lxc.namespace.time.offset.monotonic</option>
+          </term>
+          <listitem>
+            <para>
+	    Specify a positive or negative offset for the montonic clock. The
+	    format accepts hours (h), minutes (m), seconds (s),
+	    milliseconds (ms), microseconds (us), and nanoseconds (ns).
+            </para>
+          </listitem>
+        </varlistentry>
```
Comment (stgraber) [inline doc/lxc.container.conf.sgml.in:1834]: Should be its own commit, no?

File: src/lxc/confile.c (hunk 1 above)
Comment (brauner) [reply src/lxc/confile.c:227]: Yeah, I'm kinda torn. I went back and forth. But yeah, we could.

Comment (stgraber) [reply src/lxc/confile.c:227]: `lxc.namespace` seems about namespacing sharing/attaching. Normal namespace configuration has traditionally been done at the toplevel (`lxc.mount`, `lxc.net`, `lxc.idmap`, ...)

File: src/lxc/confile.c (hunk 2 above)
Comment (brauner) [reply src/lxc/confile.c]: This looks like something that github is displaying weirdly because I'm sure not seeing that locally.

**Comment by botieking98** [comment]
@brauner How is it work?
